src/ext/autodoc_traits/autodoc_traits.py

```diff
@@ -0,0 +1,107 @@
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Autodoc extension for configurable traits.
+This code auto documents traits from Cylc UI Server:
+Acknowledgment:
+Code derived from the Jupyter Hub source (BSD).
+
+https://github.com/jupyterhub/autodoc-traits/
+    autodoc_traits.py
+
+BSD 3-Clause License
```

Comment:
@dpmatthews, I have added the License in its entirety, with a note it has been modified and a link to the original code.

Reply:
Thanks - looks good to me